### PR TITLE
[Merged by Bors] - enhancement(console sink): Improve error handling

### PR DESCRIPTION
**Pull request by juchiast**
- Don't panic.
- Emit a warning on missing field, previously it just prints an empty line.

Signed-off-by: Duy Do <juchiast@gmail.com>

### CODE REVIEW

File: src/sinks/console.rs [hunk 1]
```diff
@@ -124,9 +123,19 @@ struct WriterSink {
 impl StreamSink for WriterSink {
     async fn run(&mut self, mut input: BoxStream<'_, Event>) -> Result<(), ()> {
         while let Some(event) = input.next().await {
-            write_event_to_output(&mut self.output, event, &self.encoding)
+            let buf = match encode_event(event, &self.encoding) {
+                Some(mut buf) => {
+                    buf.push('\n');
+                    buf
+                }
+                None => continue,
```
Comment (fanatid) [inline src/sinks/console.rs:131]: `self.acker.ack(1)` should be called anyway

**Review by fanatid — CHANGES_REQUESTED** [review]

File: src/sinks/console.rs (hunk 1 above)
Comment (juchiast) [reply src/sinks/console.rs]: moved `ack` to the top.

**Comment by fanatid** [comment]
What will happen with transforms connected to sink if sink will finish with error? They will stop produce items because the pipeline will be full? Can sinks return a result if no signal outside? (It's more about topology, maybe @ktff will able to answer)

**Comment by juchiast** [comment]
@fanatid A test shows that vector just stops if `console` sink return Err.

```
Oct 03 15:58:15.112 ERROR sink{topology_component_name=console topology_component_type=console}: vector::topology: An error occurred that vector couldn't handle.
```

**Review by fanatid — APPROVED** [review]
lgtm
@juchiast can we wait Luke or Kruno answer regarding no panic behavior? I do not remember that any other sinks return error result.

**Comment by juchiast** [comment]
@fanatid yes we should wait! 

Btw why does the second commit only trigger 5 checks, seem weird.

**Comment by fanatid** [comment]
Strange :thinking: any ideas @jamtur01?

**Comment by jamtur01** [comment]
@fanatid because we turned on Bors. When you're ready to merge follow the instructions in contributing.

**Comment by ktff** [comment]
> What will happen with transforms connected to sink if sink will finish with error?

When any sink panics or finishes with an error we start a graceful shutdown of the process.

**Comment by ktff** [comment]
So replacing panic with a returned Error, doesn't change anything regarding topological behavior, although returning error is preferred.

**Comment by juchiast** [comment]
bors merge

**Pull request closed without merging**